crates/router/src/connector/square/transformers.rs

```diff
@@ -0,0 +1,209 @@
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+use crate::{core::errors,types::{self,api, storage::enums}};
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct AmountMoneyType {
+    amount: i64,
+    currency: String
+}
+
+//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+pub struct SquarePaymentsRequest {
+    amount_money: AmountMoneyType,
+    idempotency_key: String,
+    source_id: String,
+}
+
+// i am using statement_descriptor_suffix as a means to supply card token to the request, as create-payment on square seems to use tokens instead of payment options
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for SquarePaymentsRequest  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        let payment_request = Self {
+            amount_money: AmountMoneyType {
+                amount: item.request.amount,
+                currency: item.request.currency.to_string()
+            },
+            idempotency_key: Uuid::new_v4().to_string(),
+            source_id: match &item.request.statement_descriptor_suffix {
+                Some (val) => String::from(val),
+                None => String::from("cnon:card-nonce-ok")
+            }
+        };
+        Ok(payment_request)
+    }
+}
```

Comment:
@bernard-eugine Have a look at this. I don't think we can use that field for this purpose

Reply:
Are we referring to source_id field? Why us "cnon:card-nonce-ok" being passed for None case?